fearless/Common/PurchaseProvider/MoonpayProvider.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import FearlessUtils
+import IrohaCrypto
+
+final class MoonpayProvider: PurchaseProviderProtocol {
+    static let baseUrlString = "https://buy.moonpay.com/"
+
+    private var colorCode: String?
+    private var callbackUrl: URL?
+
+    let apiKey: String
+
+    init(apiKey: String) {
+        self.apiKey = apiKey
+    }
+
+    var hmacSigner: HmacSignerProtocol?
+
+    func with(colorCode: String) -> Self {
+        self.colorCode = colorCode
+        return self
+    }
+
+    func with(callbackUrl: URL) -> Self {
+        self.callbackUrl = callbackUrl
+        return self
+    }
+
+    func buildPurchaseAction(
+        for chain: Chain,
+        assetId _: WalletAssetId?,
+        address: String
+    ) -> [PurchaseAction] {
+        let optionUrl: URL?
+
+        switch chain {
+        case .polkadot:
+            optionUrl = buildURLForToken("DOT", address: address)
+        case .kusama:
+            optionUrl = nil
+        case .westend:
+            optionUrl = nil
+        }
+
+        if let url = optionUrl {
+            let action = PurchaseAction(title: "MoonPay", url: url, icon: R.image.iconMoonPay()!)
```

Comment:
force unwrap icon. Let icon be an optional value

Reply:
It's guaranteed that icon is always there. If it's not, the project will not build